frontend/public/components/utils/dom-utils.tsx

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+
+const isHTMLElement = (n: Node): n is HTMLElement => {
+  return n.nodeType === Node.ELEMENT_NODE;
+};
+
+export const getParentScrollableElement = (node: HTMLElement) => {
+  let parentNode: Node = node;
```

Comment:
i like the idea a lot here and would potentially like to add this helper in PF as well (walking the parent nodes until we find a scrollable element). Can we possibly accept a query string argument as well, maybe `scrollableElement`, that allows us to explicitly select a scrollable element in some cases that we don't want to find it automagically?

Reply:
@priley86 I added a prop to table to allow specifying a scroll element or a callback that returns one. Let me know if this works. I don't particularly like using selectors with react. But the callback API would allow someone to use a selector if they needed:
`<Table scrollElement={() => document.getElementById('content-scrollable')}`

Reply:
i'm fine with this. It looks like you are just hitting some test failures. Nice improvement though!

Reply:
whoops. Forgot to make `scrollElement` optional.

Reply:
@christianvogt, there appears to be a bug here.  Adding `scrollElement` to `<Table>` as described in https://github.com/openshift/console/pull/2631/commits/5f4351c894571526cf704ff9e83bac71f6bb322b#r322474613 results in a typescript error:
```
ERROR in /Users/rhamilto/Git/console/frontend/public/components/pod.tsx(320,44):
TS2322: Type '{ aria-label: string; Header: { (): ({ title: string; sortField: string; transforms: ((label: string, extra: IExtra) => ISortable)[]; props: { className: string; }; sortFunc?: undefined; } | { title: string; sortFunc: string; transforms: ((label: string, extra: IExtra) => ISortable)[]; props: { ...; }; sortField?: u...' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<Component<Pick<any, never> & TableProps, any, any>> & Readonly<Pick<any, never> & TableProps> & Readonly<{ children?: ReactNode; }>'.
  Property 'scrollElement' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<Component<Pick<any, never> & TableProps, any, any>> & Readonly<Pick<any, never> & TableProps> & Readonly<{ children?: ReactNode; }>'.
```

Reply:
@rhamilto looks like `TableProps` is missing the `scrollElement` type?

Reply:
Yup.  But it also looks like the types are wrong.  
```
ERROR in /Users/rhamilto/Git/console/frontend/public/components/pod.tsx(417,5):
TS2322: Type 'Element' is not assignable to type 'HTMLElement | (() => HTMLElement)'.
  Type 'Element' is missing the following properties from type 'HTMLElement': accessKey, accessKeyLabel, autocapitalize, dir, and 110 more.
```

Reply:
Probably safe to make this change too: `HTMLElement` => `Element`